### PR TITLE
Add an article about UI testing with new Xcode and resetting data

### PR DESCRIPTION
**Pull request by TomasLinhart**
Please review my English and also content of the article. Thanks.

### CODE REVIEW

File: _posts/2015-10-27-resetting-data-after-each-ui-test.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+But as you can see, you can set `launchArguments` which allows you to set custom arguments that can be read when the application starts and based on that, the application can remove its content and settings. So we can easily pass an argument to the `setUp()` method.
+
+```swift
+let application = XCUIApplication()
+application.launchArguments = ["--ResetData"]
+application.launch()
+```
+
+After that we need to add a code before `UIApplicationMain` is executed to reset the data. Unfortunately, there is no main.swift on iOS where you could add this code and you have only an annotation on AppDelegate with `@UIApplicationMain`. Fortunately, you can remove this annotation and create a file main.swift where you place a code that you would have in your Objective-C project.
```
Comment (maximevaly) [inline _posts/2015-10-27-resetting-data-after-each-ui-test.md]: I'd says "to add code" or "to add some code", and main.swift would look nicer as `main.swift`.

File: _posts/2015-10-27-resetting-data-after-each-ui-test.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+```swift
+autoreleasepool {
+    UIApplicationMain(Process.argc, Process.unsafeArgv, nil, NSStringFromClass(AppDelegate))
+}
+```
+
+And before that you can easily put a code for resetting the data that you will want to enable only in the debug mode. To enable the DEBUG flag, you need to add to your project settings a value under `Other Swift Flags`. You put `-D DEBUG` for Debug.
+
+```swift
```
Comment (maximevaly) [inline _posts/2015-10-27-resetting-data-after-each-ui-test.md]: :+1: the syntax coloration

File: _posts/2015-10-27-resetting-data-after-each-ui-test.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+After that we need to add a code before `UIApplicationMain` is executed to reset the data. Unfortunately, there is no main.swift on iOS where you could add this code and you have only an annotation on AppDelegate with `@UIApplicationMain`. Fortunately, you can remove this annotation and create a file main.swift where you place a code that you would have in your Objective-C project.
+
+```swift
+autoreleasepool {
+    UIApplicationMain(Process.argc, Process.unsafeArgv, nil, NSStringFromClass(AppDelegate))
+}
+```
+
+And before that you can easily put a code for resetting the data that you will want to enable only in the debug mode. To enable the DEBUG flag, you need to add to your project settings a value under `Other Swift Flags`. You put `-D DEBUG` for Debug.
```
Comment (maximevaly) [inline _posts/2015-10-27-resetting-data-after-each-ui-test.md]: Same as before, "code", "some code", "a line of code".

File: _posts/2015-10-27-resetting-data-after-each-ui-test.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
+---
+layout: post
+title: Resetting the application data after each test with Xcode 7 UI Testing
+permalink: /resetting-application-data-after-each-test-with-xcode7-ui-testing
+tags: ios, swift, xcode, testing, ui-testing
+categories: ios
+author: Tomáš Linhart
```
Comment (maximevaly) [inline _posts/2015-10-27-resetting-data-after-each-ui-test.md]: @dirkaholic just proposed an update in PR #7 so you might need to rebase, update the new "_data/authors.yml" file and change that to `author_id:`.

**Comment by maximevaly** [comment]
Looks good and useful but I can't really judge the code, @pmiaskiewicz can you please do a review?

File: _posts/2015-10-27-resetting-data-after-each-ui-test.md (hunk 1 above)
Comment (TomasLinhart) [inline _posts/2015-10-27-resetting-data-after-each-ui-test.md]: main.swift is not a problem, what do you think about **add the following lines**?

File: _posts/2015-10-27-resetting-data-after-each-ui-test.md (hunk 4 above)
Comment (dirkaholic) [inline _posts/2015-10-27-resetting-data-after-each-ui-test.md]: #7 is merged so now you have to :grinning:

File: _posts/2015-10-27-resetting-data-after-each-ui-test.md (hunk 1 above)
Comment (maximevaly) [inline _posts/2015-10-27-resetting-data-after-each-ui-test.md]: :+1:

**Comment by TomasLinhart** [comment]
@maximevaly I have rebased and modified things that you suggested.

**Comment by pmiaskiewicz** [comment]
:+1:

**Comment by osgafarov** [comment]
:+1:

**Pull request merged**